packages/perseus-editor/src/components/util.ts

```diff
@@ -253,3 +260,37 @@ export const getQuadraticEquation = (startCoords: [Coord, Coord, Coord]) => {
         "y = " + a.toFixed(3) + "x^2 + " + b.toFixed(3) + "x + " + c.toFixed(3)
     );
 };
+
+export const shouldShowStartCoordsUI = (flags, graph) => {
+    // TODO(LEMS-2228): Remove flags once this is fully released
+    const startCoordsUiPhase1Types = [
+        "linear",
+        "linear-system",
+        "ray",
+        "segment",
+        "circle",
+    ];
+    const startCoordsUiPhase2Types = ["sinusoid", "quadratic"];
+
+    const startCoordsPhase1 = flags?.mafs?.["start-coords-ui-phase-1"];
+    const startCoordsPhase2 = flags?.mafs?.["start-coords-ui-phase-2"];
+    const startCoordsPoint = flags?.mafs?.["start-coords-ui-point"];
+
+    if (startCoordsPhase1 && startCoordsUiPhase1Types.includes(graph.type)) {
+        return true;
+    }
+
+    if (startCoordsPhase2 && startCoordsUiPhase2Types.includes(graph.type)) {
+        return true;
+    }
+
+    if (
+        startCoordsPoint &&
+        graph.type === "point" &&
+        graph.numPoints !== "unlimited"
+    ) {
+        return true;
+    }
+
+    return false;
+};
```

Comment:
Moved the flag logic out into the util file because it was getting super cluttered inside the interactive-graph-editor.tsx.

Reply:
👍 It's nice to have a place for this logic!